.github/workflows/dotnet-build-and-test.yml

```diff
@@ -20,57 +20,90 @@ permissions:
   contents: read
 
 jobs:
+  paths-filter:
+    runs-on: ubuntu-latest
+    outputs:
+      dotnetChanges: ${{ steps.filter.outputs.dotnet }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            dotnet:
```

Comment:
I need to test/look up the documentation for this. I'm assuming it's a logical or of the conditions. But I don't actually know that.